packages/hardhat-core/src/internal/core/config/config-validation.ts

```diff
@@ -573,3 +574,17 @@ export function getValidationErrors(config: any): string[] {
   const ioTsErrors = DotPathReporter.report(result);
   return [...errors, ...ioTsErrors];
 }
+
+export function validateResolvedConfig(resolvedConfig: HardhatConfig) {
+  console.log(resolvedConfig.solidity.compilers);
+  const soliditySettings: any[] = resolvedConfig.solidity.compilers.map(
+    (s) => s.settings
+  );
+  const optimiser: any[] = soliditySettings.map((o) => o.optimizer);
+  const runs: number[] = optimiser.map((r): number => r.runs);
```

Comment:
I believe this will throw if there are no optimizer settings. We should use a single map that does something like `settings?.optimizer?.runs` and returns `Array<number | undefined>`, and then we should handle the possible undefined value.

Reply:
I have checked and if we don't add the optimizer settings then the solidity compiler adding default values so I don't think that this will throw any error

Reply:
The compiler will add default values when you execute `solc`, but this is Hardhat's config, that can have empty settings.